### PR TITLE
WIP: Add mapgen diamond-square

### PR DESCRIPTION
**Pull request by vladtcvs**
Generate beatiful mountains using diamond-square algorythm: #11213 

This PR is a Work in Progress

### CODE REVIEW

**Comment by vladtcvs** [comment]
image

**Comment by vladtcvs** [comment]
image

File: src/mapgen/diamond_square/diamond_square.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+	double base;
+};
+
+class MountainLandscape
+{
+    PseudoRandom *prandom;
+    public:
+        MountainLandscape(PseudoRandom *prandom) {this->prandom = prandom;}
+        std::list<std::tuple<DiamondSquareMountain *, std::pair<double, double>>> mountains;
```
Comment (sfan5) [inline src/mapgen/diamond_square/diamond_square.h:37]: small bit of advice: We have a `core::vector2d<double>` type you could use here,
perhaps `core::vector2d<float>` (short `v2f`) would also suffice.

**Comment by Emojigit** [comment]
This mapgen cannot generate good plains

**Comment by ThatElektrika** [comment]
wow...

**Comment by rubenwardy** [comment]
Closing as WIP with no progress. Also, is not on the roadmap

**Pull request closed without merging**